### ISSUE TITLE
Support a `main` branch

### ISSUE THREAD

**Issue by steven-terrana** [jenkins-infra/plugin-site#630]
Hello! 

Thank you for the work that you do maintaining jenkins infra. 

The change to pull in `README.md` documentation was fantastic and was working as expected while our project ([Jenkins Templating Engine](https://github.com/jenkinsci/templating-engine-plugin)) utilized a `master` branch. 

We have updated our project's "trunk" branch to be called `main` and it seems to have broken the integration with the plugin site. 

I'd be more than happy to try and resolve if you could point me in the right place to look :)

**Comment by halkeye** [comment]
aww man, kinda frustrating timing.

tldr; So easiest right now fix, is to make your url in your pom.xml to point to the raw file instead of the repo

longer version:

We just finished updates to update-center to publish the default branch, you can see that in https://updates.jenkins.io/current/update-center.actual.json
So after that was finished a couple weeks ago, I was going to update https://github.com/jenkins-infra/plugin-site-api/search?q=master to pull that info and use it, but realized that plugin-site-api is a bit of a mess over time, so we/i am going to split apart the runtime vs the compile time parts of plugin-site-api. I expect I'll be able to get this done over the weekend, and then the logic to find the master/main/etc branch will be entirely in plugin-site and easier to update.

So if your happy to wait a week or so, then don't do anything, other than maybe switch back from main to master.
If you want it sooner, you can try and update plugin-site-api and roll it out

**Comment by steven-terrana** [comment]
Appreciate the work around and the context. 

i'm happy to wait! not urgent, folks will get by for the time being :)

**Comment by oleg-nenashev** [comment]
> So easiest right now fix, is to make your url in your pom.xml to point to the raw file instead of the repo

It is a good workaround.
The fallback behavior is not terrible either, so users might be OK with waiting a bit.

![image](https://user-images.githubusercontent.com/3000480/113937045-e388a980-97f8-11eb-9bbe-c2eacbab4b19.png)

**Comment by halkeye** [comment]
So porting the logic to plugin-site directly was too overwhelming, so managed to get it in https://github.com/jenkins-infra/plugin-site-api/pull/98

Interesting note, the original template engine works fine because github redirects requests to master to main after you rename. But isn't super consistent.
